lib/src/native/realm_core.dart

```diff
@@ -1096,8 +1108,12 @@ class _RealmCore {
     });
   }
 
-  String userGetCustomData(User user) {
-    final customDataPtr = _realmLib.invokeGetPointer(() => _realmLib.realm_user_get_custom_data(user.handle._pointer));
+  String? userGetCustomData(User user) {
+    final customDataPtr = _realmLib.realm_user_get_custom_data(user.handle._pointer);
+    if (customDataPtr == nullptr) {
```

Comment:
You can use `toRealmDartString` to handle the null handling as well as the freeing.

Reply:
goog point. will redo